src/services/predictionLayerService.tsx

```diff
@@ -21,79 +21,95 @@ export function addPolygonLayer(map: mapboxgl.Map, aoi: Polygon) {
         },
     })
 }
+export async function addPredictionLayer(map: mapboxgl.Map, datetime: number | undefined, regionId: number | undefined): Promise<any> {
+    if (!datetime || !regionId) {
+        console.error('Datetime', datetime)
+        console.error('RegionId', regionId)
+        return []
+    }
+
+    //timout to simulate loading
+    //await new Promise((resolve) => setTimeout(resolve, 1000))
+
+    const predictions = await getJobPredictions(datetime, regionId)
+
+    //check if source already exists
+    if (map.getSource(`pred-${datetime}:${regionId}`)) {
+        map.removeLayer(`pred-${datetime}:${regionId}`)
+        map.removeSource(`pred-${datetime}:${regionId}`)
+    }
```

Comment:
to me this does not make sense, this means if I want to add a layer that already exists we are removing the layer so that we can add it again?